src/queryIterator.ts

```diff
@@ -15,7 +16,7 @@ import { Response } from "./request/request";
  * traversal and iterating over the response
  * in the Azure Cosmos DB database service.
  */
-export class QueryIterator<T> {
+export class QueryIterator<T = Resource & ItemDefinition> {
```

Comment:
@christopheranderson how about this instead:

`QueryIterator<T = Partial<Resource> & ItemDefinition>`

Reply:
Even Partial isn't correct because I can select a concat of properties into a new property. :(

And in this case, Query Iterator can be of type ContainerDefintion/etc. as well, so this really won't work at all. This has to stay T, and then fixed in Items.query where it passes the type along.